packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -29,7 +30,7 @@ var publicPath = '/';
 // `publicUrl` is just like `publicPath`, but we will provide it to our app
 // as %PUBLIC_URL% in `index.html` and `process.env.PUBLIC_URL` in JavaScript.
 // Omit trailing slash as %PUBLIC_PATH%/xyz looks better than %PUBLIC_PATH%xyz.
-var publicUrl = '';
+var publicUrl = ensureSlash(paths.servedPath, false);
```

Comment:
I'm not really sure why this was changed, but I know it still worked after. I didn't test HMR, though -- I'll remove it as I don't know why the homepage is relevant to the dev environment.